### PR TITLE
Re-skip CheckpointIO tests

### PR DESCRIPTION
**Pull request by jwpeterson**
This snuck in without proper testing along with #9812.

### CODE REVIEW

**Review by permcody — APPROVED** [review]
Note: I know this is a revert, but I tested on my system and the .cpr tests are all passing on the Mac. This issue only affects the .cpa files.

**Comment by permcody** [comment]
Emergency Fix: Merging

**Comment by roystgnr** [comment]
Wait, what?  It only affects the *ASCII files*??  https://www.moosebuild.org/job/114922/ seems to show the opposite, which is what I would have expected.

**Comment by roystgnr** [comment]
@permcody, could you give me a stack trace from a devel or dbg run?  The Civet results make it look as if data_size itself isn't being read correctly in the non-ascii cases, but that's literally the first datum after the version string, and it's hard-coded to uint64_t, which I'm 99.9% sure we can use via cross-platform XDR...

**Comment by permcody** [comment]
Let me rephrase - The ASCII tests are the ones that are broken. Building the debug target now. I'll post more information shortly.

**Comment by permcody** [comment]
As it is in the test spec:
```
mpiexec -n 2 /Users/permcj/projects/moose/test/moose_test-devel --n-threads=1 -i checkpoint_split.i Outputs/file_base=test_2  Mesh/file=checkpoint_split_in.2.cpr --error --no-gdb-backtrace


[0] ../src/mesh/checkpoint_io.C, line 593, compiled nodate at notime

[1] ../src/mesh/checkpoint_io.C, line 593, compiled nodate at notime




*** ERROR ***



application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
```

Running on a single processor:
```
mpiexec -n 1 /Users/permcj/projects/moose/test/moose_test-devel --n-threads=1 -i checkpoint_split.i Outputs/file_base=test_2  Mesh/file=checkpoint_split_in.2.cpr --error --no-gdb-backtrace


Stack frames: 15
0: 0   libmesh_devel.0.dylib               0x000000011115c3e2 libMesh::print_trace(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) + 1506
1: 1   libmesh_devel.0.dylib               0x00000001111577aa libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*) + 330
2: 2   libmesh_devel.0.dylib               0x0000000111614ae1 libMesh::CheckpointIO::read(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) + 4689
3: 3   libmesh_devel.0.dylib               0x00000001118618a1 libMesh::NameBasedIO::read(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) + 5617
4: 4   libmesh_devel.0.dylib               0x00000001118c2bf2 libMesh::UnstructuredMesh::read(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, void*, bool, bool) + 258
5: 5   libmoose-devel.0.dylib              0x00000001104138c2 FileMesh::buildMesh() + 1106
6: 6   libmoose-devel.0.dylib              0x0000000110434346 MooseMesh::init() + 246
7: 7   libmoose-devel.0.dylib              0x00000001100f0eb3 SetupMeshAction::act() + 179
8: 8   libmoose-devel.0.dylib              0x00000001100b0600 ActionWarehouse::executeActionsWithAction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) + 752
9: 9   libmoose-devel.0.dylib              0x00000001100b02fb ActionWarehouse::executeAllActions() + 123
10: 10  libmoose-devel.0.dylib              0x000000011029bf5c MooseApp::runInputFile() + 252
11: 11  libmoose-devel.0.dylib              0x000000011029ce59 MooseApp::run() + 89
12: 12  moose_test-devel                    0x000000010fbc5cc4 main + 116
13: 13  libdyld.dylib                       0x00007fff948ea235 start + 1
14: 14  ???                                 0x0000000000000008 0x0 + 8
[0] ../src/mesh/checkpoint_io.C, line 593, compiled nodate at notime




*** ERROR ***



Stack frames: 7
0: 0   libmesh_devel.0.dylib               0x000000011115c3e2 libMesh::print_trace(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) + 1506
1: 1   libmoose-devel.0.dylib              0x00000001102b4609 moose::internal::mooseErrorRaw(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) + 969
2: 2   libmoose-devel.0.dylib              0x000000011029d0c1 void mooseError<char const*>(char const*&&) + 337
3: 3   libmoose-devel.0.dylib              0x000000011029cf52 MooseApp::run() + 338
4: 4   moose_test-devel                    0x000000010fbc5cc4 main + 116
5: 5   libdyld.dylib                       0x00007fff948ea235 start + 1
6: 6   ???                                 0x0000000000000008 0x0 + 8
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
```

If you want me to enable tracefiles, I can do that too.

**Comment by roystgnr** [comment]
Well, hell, it really is hitting the error in switch(data_size).

Are you sure the ASCII cases are broken?  cpa is ASCII, cpr is binary, and you posted a failure with cpr on the command line.

Could you get in there in the debugger and tell me what it thinks data_size *is*?

Looking at the binary file now and trying to compare with XDR specs...

**Comment by permcody** [comment]
Damn... I'm definitely blind. I looked at that three times and saw .cpa. You are correct, it's the .cpr cases that are broken.

**Comment by roystgnr** [comment]
Things I've learned in the last 5 minutes:

XDR appears to output even short ints as 4 bytes rather than 2, which is a standard part of the protocol and which could explain why even xda.gz tends to give us smaller files.

XDR appears to have output a uint64_t as 4 bytes, which is definitely *not* a standard part of the protocol and which makes me fear I've got a bug in the xdr_translator specializations...

**Comment by permcody** [comment]
From the debugger:
```
p input_version
(std::__1::string) $1 = "checkpoint-1.2"

p data_size
(libMesh::CheckpointIO::header_id_type) $2 = 34359738370
```

**Comment by roystgnr** [comment]
XDR, YOU HAD ONE JOB!

From the duplicitous linux XDR man page:

> bool_t xdr_u_long(XDR *xdrs, unsigned long *ulp);
>
> A filter primitive that translates between C unsigned long integers and their external representations.

But from the IBM [AIX7 XDR man pages](https://www.ibm.com/support/knowledgecenter/en/ssw_aix_72/com.ibm.aix.commtrf1/xdr_u_long.htm):

>    Note
>    The xdr_u_long subroutine encodes or decodes a 32-bit value, irrespective of whether the application is compiled in 32-bit mode or in 64-bit mode. If a 64-bit value is passed to the xdr_u_long subroutine, the resulting high-order 32-bit values are not determined.

uint_64_t is equivalent to "unsigned long" on Linux, and so my template specializations picked out the XDR unsigned long method, because the name made it seem like an appropriate method to call rather than an insane trap that should have never existed.  Presumably uint_64_t is equivalent to "unsigned long long" on OSX, and XDR is at least sane enough to assume that "long long" implies 8 bytes - OSX then tries to read both the 4 bytes containing the value we intended it to read and the 4 bytes afterward, and ends up with a huge number after interpreting the intended low-order bits as high-order bits.  (also, ends up with a broken offset into the file)

This is pretty awful.  But how have we not hit it before?  I'd assume the same bug would affect XDR file compatibility between Linux and Mac *whenever* we saved 8-byte unique_id values, and those are enabled by default with MOOSE, right?

**Comment by roystgnr** [comment]
Huh, no they're not.  Looks like unique_id() is only required for XFEM (and implicitly, for solution I/O with overlapping nodes, I guess).  For some reason I thought it was exercised more extensively than that.

**Comment by permcody** [comment]
Well we've long wanted to take advantage of the unique ID system to help us deal with Material properties with repartitioning. We've yet to tackle that issue but when we get there, we'll need unique IDs turned on all the time in MOOSE.
